### DISCUSSION TITLE
Fetching data trough join table doesn't work

### DISCUSSION THREAD

**Discussion by kharalanova** [supabase/supabase#12370] [Questions] [answered]
I am trying to fetch some data with this ReactNative code:

```
 async getTripServices(tripId: number) {
    const query = `title, 
            difficulty, 
            trips_services(
              services(
                name
              )
            )`;
    return supabase.from("trips").select(query).eq("id", tripId);
```

The name from table services doesn't get populated in the response I receive. As you can see in the response I get **"services: null"**

**I get this response:**
Trip Services Response
```
Array [
  Object {
    "difficulty": "средна",
    "title": "Триград и Ягодина",
    "trips_services": Array [
      Object {
        "services": null,
      },
      Object {
        "services": null,
      },
    ],
  },
]
```

**My tables:** 
```
trips (
  id primary key int8,
  title varchar,
  difficulty varchar
 .....
)
services (
  id primary key int8,
  name varchar,
 ....
)
```

join table 
```
trips_services (
 id primary key int8,
 trip_id int8 not null references trips,
 service_id int8 not null references services
)
```

**Comment by GaryAustin1** [comment, marked as answer]
Make sure trips_services has both foreign keys as part of the primary key (required by PostgREST the API for many-many table).
Then don't use it in the select.  Just use `('...., services(name)')`.

**Reply by kharalanova** [reply]
Hi GaryAustin1, 
Thank you for your answer. What you have suggested  I already tried. It was how I first implemented the trip_services table and it didn't make any difference to the outcome:

I defined trips_services with a composite primary key as suggested in the documentation
```
trips_services (
    primary key (trip_id, service_id),
    trip_id int8 not null references trips,
    service_id int8 not null references services,
)
```

Then I run 
```
 const query = `title, 
            difficulty, 
              services(
                name
              )`;
    return supabase.from("trips").select(query).eq("id", tripId);
```

and I got this outcome:
```
Array [
  Object {
    "difficulty": "средна",
    "services": Array [],
    "title": "Триград и Ягодина",
  },
]
```

In this variation of the code, it seems even the fact that there are actually 2 services to include is not picked up

**Reply by GaryAustin1** [reply]
Could be you are not meeting the RLS select policies on all the other tables involved.

**Reply by kharalanova** [reply]
Thank you GaryAustin1, 

This was very helpful comment. I never intended to add RLS policies to these tables, but for some reason:
  - **trips** table was defined as RLS disabled 
  - **services** table as RLS enabled and No policies added. 
This was causing the issue.

Once I changed **services** to RLS disabled, it resolved my issue.
Much appreciated
Katrin
